### ISSUE TITLE
ENH: a readable sample function or an improvement

### ISSUE THREAD

**Issue by nagaprakashv** [pandas-dev/pandas#54227]
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

Hi All! We have efficient sampling options already and it's also possible to increase "display.max_colwidth" for readability.
But still, there are many situations, where those who work with text data needs a bit more readability. Look at the following screenshot:-
![image](https://github.com/pandas-dev/pandas/assets/13671867/d990e7ee-a60c-42df-bd18-f8297785fd06)

### Feature Description

I've been doing EDA and also into QA of datasets for the past 5+ years, and spot checks, etc. are impossible without readability.

I've been using this simple, skeletal for loop function for a while. This is really useful. Can we improve on this?
![image](https://github.com/pandas-dev/pandas/assets/13671867/b54955fb-de78-4e41-8764-4a0b8b01edb8)

`def sample(df):`
`    row = df.sample()`
`    for i in row:`
`        print(i)`
`        print("", row[i].iloc[0], '\n')`

### Alternative Solutions

The only customization is display.max_colwidth, and pd.options.display.max_rows is not related to this for now.

**Comment by jesse-sealand** [comment]
What would the enhancement provide? Would this be like transposing each row into the format you provided?

**Comment by nagaprakashv** [comment]
> What would the enhancement provide? Would this be like transposing each row into the format you provided?

Yes! It's convenient for those who like to read/compare all the text in the cells and do it a lot regularly. So a single function will reduce their effort, making it a good exploratory process.

**Comment by mroeschke** [comment]
Thanks for the suggestion but given the lack of interest from the core devs and community, closing
